### PR TITLE
Link google account client

### PR DESCRIPTION
**Pull request by prastoin**
close https://github.com/AdonisEnProvence/MusicRoom/issues/421

### CODE REVIEW

File: packages/client/screens/MySettings/settingsMachine.ts [hunk 1]
```diff
@@ -323,6 +342,114 @@ export const settingsMachine =
...
+                                    always: [
+                                        {
+                                            cond: 'User has already linked a google account',
+                                            target: 'User has linked a google account successfully',
+                                        },
+                                        {
+                                            target: 'Waiting for user to start a link google account operation',
+                                        },
+                                    ],
+                                },
```
Comment (Devessier) [inline packages/client/screens/MySettings/settingsMachine.ts:359]: **👏 praise:** ‏I like this choose state!

File: packages/client/screens/MySettings/settingsMachine.ts [hunk 2]
```diff
@@ -323,6 +342,114 @@ export const settingsMachine =
...
+
+                                                actions:
+                                                    'Display toast error link google account server error google id unavailable',
+
+                                                target: "#Settings.Fetched user's settings.Link Google account manager.Waiting for user to start a link google account operation",
+                                            },
+                                            {
+                                                actions: [
+                                                    'Display link google account operation success toast',
+                                                    'reset user google access token from context',
```
Comment (Devessier) [inline packages/client/screens/MySettings/settingsMachine.ts:430]: **🎯 suggestion:** ‏I think there is a typo. The action should be _Reset user google access token from context_. (The first _r_ is capitalized.)

**Review by Devessier — APPROVED** [review]
Nice job! Nice use of XState in the application and in tests.
I made a little suggestion.

File: packages/client/screens/MySettings/settingsMachine.ts (hunk 2 above)
Comment (prastoin) [reply packages/client/screens/MySettings/settingsMachine.ts:430]: Ah nice one, thx about to be updated !

**Review by Devessier — APPROVED** [review]
Nice!
We can merge.

**Review by Devessier — APPROVED** [review]
We can merge.

**Pull request merged**